[scripts/get] makes sudo an optional dependency

tonyfahrion opened the pull request:
If the execution user is already the root user, this avoids requiring the 
sudo command to be installed within the environment.

Which might help in CI environments like alpine.

Tested with `ubuntu:latest` and `alpine:latest`

Commit:
makes sudo an optional dependency in scripts/get

If the execution user is already the root user, this avoids requireing
sudo to be installed within the environment. Which might helo in CI
environments.

Commit:
fix scripts/get runAsRoot CMD var building

Avoid using $* multiple times

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/helm/2626)
<!-- Reviewable:end -->

thomastaylor312 commented on scripts/get:
```diff
@@ -46,6 +46,17 @@ initOS() {
+runAsRoot() {
+  local CMD="$*"
+  
+  if ! whoami | egrep -q '^root$'; then
```
If I remember my bash correctly, I think the best way to do this is to check if `$EUID` is equal to 0. According to [this SO answer](https://stackoverflow.com/a/18216122), `EUID` is read only and the best way to make sure it is a root user.

twolfy replied:
Thx! Very good point! Didn't know that `$EUID` exists (`env` doesn't show it).

I changed the PR via 8eace38, please have a look

thomastaylor312 reviewed (changes requested):
This is a great patch! I just ran into this the other day. One change is needed on how you are checking for root and then it should be good

Commit:
improve check if we are root already - in scripts/get

$EUID is readonly, so it should be save to test against it.

Thanks for the tip @thomastaylor312 !

technosophos reviewed (approved):
LGTM

technosophos commented:
I'll let @thomastaylor312 make the final call on it, but this is looking good.

technosophos commented:
Also, IMHO this meets the "stability/security" requirement to be added to 2.6.

thomastaylor312 commented:
@technosophos I second that opinion. Though technically this will be available as soon as we merge because the install instructions `curl` the script from master 🙂
